src/components/layout/RightSectionWallet.vue

```diff
@@ -21,25 +21,50 @@
         {{ address }}
       </div>
       <div class="not-bootstrap-row">
-        <AeAmount :amount="balance" />
         <Dropdown
-          v-if="currencyDropdownOptions.length"
+          v-if="hasContractV2Address"
+          :options="tokenBalancesOptions"
+          :method="option => showCurrencyDropdown = !(option.token.length > 0)"
+          :selected="aeternityTokenData.token"
+        >
+          <template #displayValue="{ displayValue }">
+            <AeAmount
+              class="not-bootstrap-row"
+              :amount="displayValue.balance"
+              :token="displayValue.token"
+            />
+          </template>
+          <template v-slot="{ option }">
+            <TokenAvatarAndSymbol :address="option.token" />
+          </template>
+        </Dropdown>
+        <AeAmount
+          v-else
+          :amount="balance"
+        />
+        <Dropdown
+          v-if="currencyDropdownOptions.length && showCurrencyDropdown"
           :options="currencyDropdownOptions"
-          :method="updateCurrency"
+          :method="({ currency }) => updateCurrency(currency)"
           :selected="selectedCurrency"
-          rounded
-        />
+          show-right
+        >
+          <template #displayValue>
+            <span class="currency-value spaced">
```

Comment:
the `spaced` is only needed because, for some reason, the Dropdown displays its value inside of the flexbox

Reply:
The value could be anything(thats how I wanted it to work). The styling is supposed to be done when you pass the slot and customize it.